packages/nimble-components/src/table/index.ts

```diff
@@ -262,6 +365,27 @@ export class Table<
         this.columnNotifiers = [];
     }
 
+    private initialize(): void {
+        if (this.isInitialized) {
+            // The table is already initialized. There is nothing more to do.
+            return;
+        }
+
+        this.isInitialized = true;
+        this.$fastController.onConnectedCallback();
+        this.updateTracker.trackAllStateChanged();
+        this.observeColumns();
+    }
+
+    private async processPendingUpdates(): Promise<void> {
+        this.initialize();
+
+        do {
+            // eslint-disable-next-line no-await-in-loop
+            await DOM.nextUpdate();
+        } while (this.updateTracker.hasPendingUpdates);
```

Comment:
In my testing, I only ever saw it loop once since we don't have any existing code that mutates the table/column configuration while any of the async functions are executing.

I'll specify an upper bound of loops (maybe 10?) and write a test to validate it.

Reply:
I've spent some time attempting to write a test like you suggested, and I'm running into the problem that I can't find a way to force the code to loop multiple times. I'm not sure it's possible. This is what I'm seeing:
- modifying a property on the table queues an update using `DOM.queueUpdate()` passing a callback to be called
- `processPendingUpdates()` calls `DOM.nextUpdate()`, which returns a promise that resolves once all the callbacks passed to `DOM.queueUpdate()` have been called
- therefore, I don't see a way a new update to be queued between the time that updates are processed and `this.updateTracker.hasPendingUpdates` gets checked -- which is the only way to force another iteration of the loop

Does my understanding seem correct to you? I'm wondering if I should really be updating the code to remove the loop and just make the function be:
```
this.initialize();
await DOM.nextUpdate();
```

Reply:
We can try and run with that what about throwing if we run into that condition?

```ts
this.initialize();
await DOM.nextUpdate();
if (this.updateTracker.hasPendingUpdates) {
    throw new Error('Expected pending updates to be resolved');
}
```

(trying to be robust / log errors because I have a sneaking suspicion that fakeAsync in Angular is going to mess some of this up somewhere in a complex text because of how it controls when the FAST DOM queue runs)